### ISSUE TITLE
Gulp 4: gulp.parallel will stop gulp immediately when one task fails

### ISSUE THREAD

**Issue by per2plex** [gulpjs/gulp#1487]
In the [docs](https://github.com/gulpjs/gulp/blob/4.0/docs/API.md#gulpparalleltasks) it is stated that gulp.parallel will still execute all tasks if an error occurs:
> When the returned function is executed, the tasks or functions will be executed in parallel, all being executed at the same time. **If an error occurs, all execution will complete.**

This currently doesn't seem to be the case as the following minimal code example will demonstrate:

```javascript
var gulp = require('gulp')

gulp._settle = false

gulp.task('error', function(done) {
  setTimeout(function() {
    done(new Error('Somethings wrong!'))
  }, 200)
})

gulp.task('willnotfinish', function(done) {
  setTimeout(function() {
    console.log('I will never finish :(')
    done()
  }, 2000)
})

gulp.task('watch', function() {
  gulp.watch('watchme.js', gulp.parallel('default'))
})

gulp.task('default', gulp.parallel(
  'error',
  'willnotfinish'
))
```
This will stop running gulp after ~200ms with the following output and the task 'willnotfinish' will never finish executing:
```
Using gulpfile ~/Desktop/development/gulp-parallel-error/gulpfile.js
Starting 'default'...
Starting 'error'...
Starting 'willnotfinish'...
'error' errored after 204 ms
Error: Somethings wrong!
  at null._onTimeout (/Users/perplex/Desktop/development/gulp-parallel-error/gulpfile.js:5:10)
  at Timer.listOnTimeout (timers.js:92:15)
'default' errored after 211 ms
The following tasks did not complete: willnotfinish
Did you forget to signal async completion?
```

EDIT: Include watch task in code sample

**Comment by per2plex** [comment]
This actually seems to be solved by setting the `UNDERTAKER_SETTLE` environment variable to `true`, is there a reason `bach.settleParallel` is not used by default?

**Comment by per2plex** [comment]
Works with #1488 and `_settle` set to `true`. Also seems to work fine with my project which incorporates several gulp plugins, so i'm still unsure about the side effects of `_settle = true` if there are any.

I updated the code sample above to include a watch task.

**Comment by phated** [comment]
@per2plex I think the docs need to be updated because we use `process.exit` to exit with the correct code when an error occurs in a `gulp.parallel` call.  The callback gets called immediately if there is an error, and instead of waiting for all async work to be finished (the default in undertaker/bach) we immediately exit with https://github.com/gulpjs/gulp-cli/blob/master/lib/versioned/%5E4.0.0-alpha.2/index.js#L69-L71

**Comment by per2plex** [comment]
@phated I see, actually it makes sense for this to be the default behaviour. Some kind of option for  `gulp.parallel` and `gulp.series` to keep running after an error occurs in one of its tasks would be nice though.

**Comment by phated** [comment]
@per2plex we already have that command line option. Just use the --continue flag

**Comment by per2plex** [comment]
@phated Thanks, `--continue` was exactly what I was searching for, you can close this issue if it's no longer needed :)

**Comment by phated** [comment]
@ulrichb it's not something we want people to use but you can "hack the system" by setting `gulp._settle = true` before calling `gulp.parallel` (because gulp is just an instance of Undertaker and that's the property undertaker looks for).  You could also have a separate "settle" instance by doing:
```js
var settled = new gulp.Gulp();
settled._settle = true;

settled.parallel(...tasks);
```
